p2p/module.go

```diff
@@ -96,13 +120,31 @@ func (m *p2pModule) Start() error {
 			telemetry.P2P_NODE_STARTED_TIMESERIES_METRIC_DESCRIPTION,
 		)
 
-	addrBook, err := ValidatorMapToAddrBook(m.p2pCfg, m.bus.GetConsensusModule().ValidatorMap())
+	currentHeight := m.bus.GetConsensusModule().CurrentHeight()
+	var (
+		addrBook typesP2P.AddrBook
+		err      error
+	)
+
+	if m.GetBus().GetPersistenceModule() == nil {
+		// we are getting called by the client and we use the "legacy behaviour"
+		// TODO (team): improve this.
```

Comment:
I think we've kind of deprecated `team` in exchange for a name or a ticket. It's team by default. We also have an `IMPROVE` tag in the `Makefile`.

Can you change this to `IMPROVE: <explanation of why we can't delete the legacy behaviour>`